### PR TITLE
Fix CVE-2022-24823 and CVE-2022-25647 Vulnerability

### PR DESCRIPTION
**Pull request by yoyofx**
#### apm-network
* io.netty:netty-codec-http2 │ CVE-2022-24823

#### apm-sniffer/apm-agent-core
* com.google.code.gson:gson | CVE-2022-25647

#### test/basse

* guava.version:guava | CVE-2018-10237

- [ ] Add a unit test to verify that the fix works.
- [x] Explain briefly why the bug exists and how to fix it.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
All dependencies versions should be updated accordingly at here, https://github.com/apache/skywalking-java/blob/main/dist-material/LICENSE#L204

File: CHANGES.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ Release Notes.
 * Add plugin to support Tomcat 10.x.
 * Save http status code regardless of it's status.
 * Upgrade byte-buddy to 1.12.13, and adopt byte-buddy APIs changes.
-
+* Upgrade gson to 2.8.9.
+* Upgrade netty-codec-http2 to 4.1.79.Final.
 #### Documentation
```
Comment (wu-sheng) [inline CHANGES.md:17]:
```suggestion
* Upgrade netty-codec-http2 to 4.1.79.Final.

#### Documentation
```

You missed one line here.

File: pom.xml [hunk 2]
```diff
@@ -88,7 +88,8 @@
         <!-- core lib dependency -->
         <bytebuddy.version>1.12.13</bytebuddy.version>
         <grpc.version>1.44.0</grpc.version>
-        <gson.version>2.8.6</gson.version>
+        <netty.version>4.1.79.Final</netty.version>
```
Comment (wu-sheng) [inline pom.xml:91]: Why do you set the netty version separately? Isn't bumping up grpc version working?

**Comment by yoyofx** [comment]
单独更新到 grpc最近版本  v1.48.1 也一样会依赖 netty-codec-http2<https://mvnrepository.com/artifact/io.netty/netty-codec-http2> 4.1.77.Final<https://mvnrepository.com/artifact/io.netty/netty-codec-http2/4.1.77.Final> ,并不能解决此问题

参考资料:

https://mvnrepository.com/artifact/io.grpc/grpc-netty/1.48.1

Vulnerabilities from dependencies:
CVE-2022-25647<https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647>

https://mvnrepository.com/artifact/io.netty/netty-codec-http2/4.1.77.Final

2022年8月4日 16:38，吴晟 Wu Sheng ***@***.***> 写道：



@wu-sheng commented on this pull request.

________________________________

In pom.xml<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fapache%2Fskywalking-java%2Fpull%2F288%23discussion_r937509836&data=05%7C01%7C%7C6e470719bbb94551840e08da75f4c4eb%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637951991387707957%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=kqpBHHe9jpfZ3Cye0sBhA8BeGvfFOyZqMVQ9YbV44qs%3D&reserved=0>:

> @@ -88,7 +88,8 @@
         
         <bytebuddy.version>1.12.13</bytebuddy.version>
         <grpc.version>1.44.0</grpc.version>
-        <gson.version>2.8.6</gson.version>
+        <netty.version>4.1.79.Final</netty.version>


Why do you set the netty version separately? Isn't bumping up grpc version working?

**Comment by wu-sheng** [comment]
We need you to update all comments in English. This is a global community, we need to make sure all global users could understand what is going on, why we change.

File: apm-protocol/apm-network/pom.xml [hunk 3]
```diff
@@ -32,6 +32,11 @@
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-codec-http2</artifactId>
+            <version>${netty.version}</version>
+        </dependency>
```
Comment (wu-sheng) [inline apm-protocol/apm-network/pom.xml:39]: You add a `netty dependency`, but you never exclude netty dependency from gRPC dependency. 
Also, grpc dependency is also existing in many other places.

image

Especially agent-core pom.xml matters.

File: apm-sniffer/optional-plugins/guava-cache-plugin/pom.xml [hunk 4]
```diff
@@ -31,7 +31,7 @@
     <modelVersion>4.0.0</modelVersion>
 
     <properties>
-        <guava-client.version>23.0</guava-client.version>
+        <guava-client.version>30.1.1-jre</guava-client.version>
```
Comment (wu-sheng) [inline apm-sniffer/optional-plugins/guava-cache-plugin/pom.xml:34]: Don't change the plugin dependency version. This is used for compiling only.

File: apm-protocol/apm-network/pom.xml (hunk 3 above)
Comment (wu-sheng) [reply apm-protocol/apm-network/pom.xml:39]: This is still not resolved, there is another dependenc of gRPC in the pom.

File: apm-protocol/apm-network/pom.xml [hunk 5]
```diff
@@ -52,11 +52,23 @@
...
         <dependency>
             <groupId>io.grpc</groupId>
             <artifactId>grpc-stub</artifactId>
             <version>${grpc.version}</version>
+            <exclusions>
+                <exclusion>
+                    <artifactId>netty-codec-http2</artifactId>
+                    <groupId>io.netty</groupId>
+                </exclusion>
+            </exclusions>
```
Comment (wu-sheng) [inline apm-protocol/apm-network/pom.xml:71]: I can't see `netty-codec-http2` in the dependency tree of this. Why do you do this?

File: apm-protocol/apm-network/pom.xml [hunk 6]
```diff
@@ -52,11 +52,23 @@
             <groupId>io.grpc</groupId>
             <artifactId>grpc-protobuf</artifactId>
             <version>${grpc.version}</version>
+            <exclusions>
+                <exclusion>
+                    <artifactId>netty-codec-http2</artifactId>
+                    <groupId>io.netty</groupId>
+                </exclusion>
+            </exclusions>
```
Comment (wu-sheng) [inline apm-protocol/apm-network/pom.xml:60]: I can't see `netty-codec-http2` in the dependency tree of this. Why do you do this?

**Comment by yoyofx** [comment]
Excluded netty-codec-http2 from gRPC dependency from apm-agent-core and apm-network  in the POM.

I can't find a another.

File: apm-protocol/apm-network/pom.xml (hunk 3 above)
Comment (yoyofx) [reply apm-protocol/apm-network/pom.xml:44]: Is resolve, another dependency of gRPC , that not dependency netty-codec-http2 . but it dependency
netty-handler-proxy, that indirect dependency netty-codec-http2 for same version.

**Review by wu-sheng — APPROVED** [review]
Wait for @kezhenxu94 confirmation.

**Comment by wu-sheng** [comment]
image

Confirmed, the agent's dependency changed.

**Pull request merged**